lib/scheduler.js

```
@@ -29,7 +29,7 @@ function Scheduler (options) {
     var self = this;
 
     self.options = {};
-    self.options.frameworkName = options.frameworkName.replace(/ /g, "-") || "mesos-framework." + uuid.v4();
+    self.options.frameworkName = (options.frameworkName || "mesos-framework." + uuid.v4()).replace(/ /g, "-");
```

Not sure why the replace was moved. There shouldn't be any space in a UUID, but in a framework name the can potentially be some... Please change it back.

---

If there is no framework name it fails with replace undefined since frameworkName is not a string, this is a simple change to make it pass no matter what, doing nothing if there's no framework name set.

---

I think it'd be ok if it fails then... The default value can't have a space char even IMO, so I guess it doesn't really make sense to add the replace there.

---

Fails as in the constructor crashes...
It's either this or change it to an actual if sentence, do you think that's better?

---

I think we should throw an error in case the `frameworkName` is null...